youtube_dl/downloader/hls.py

```diff
@@ -102,3 +102,138 @@ def real_download(self, filename, info_dict):
         })
         self.try_rename(tmpfilename, filename)
         return True
+
+class WebVttHlsFD(FileDownloader):
+    """ A downloader for HLS WebVTT subtitles. """
+
+    @staticmethod
+    def _parse_ts(ts):
+        m = re.match('(?:(?:([0-9]+):)?([0-9]+):)?([0-9]+)(?:\.([0-9]+))?', ts)
+        hrs, min, sec, msec = m.groups()
+        return 90 * (int(hrs or 0) * 3600000 + int(min or 0) * 60000 + int(sec or 0) * 1000 + int(msec or 0))
+
+    @staticmethod
+    def _format_ts(ts):
+        ts  = int(ts / 90)
+        hrs = ts / 3600000
+        ts %=      3600000
+        min = ts /   60000
+        ts %=        60000
+        sec = ts /    1000
+        ts %=         1000
+        return '%02u:%02u:%02u.%03u' % (hrs, min, sec, ts)
+
+    def real_download(self, filename, info_dict):
+        url = info_dict['url']
+        tmpfilename = self.temp_name(filename)
+
+        self.to_screen(
+            '[hlswebvtt] %s: Downloading manifest' %
+            (info_dict['id']))
+        data = self.ydl.urlopen(url).read()
```

Comment:
This won't work in case of live HLS WebVTT streams because you constantly get new subtitle segments at the same playlist URL.

It's a shame X-TIMESTAMP-MAP support patch hasn't been merged to ffmpeg yet after 3 years, but in my use-case (vlive.tv) it's not required, so dumping HLS WebVTT via ffmpeg works quite good.